yogstation/code/modules/mob/say.dm

```diff
@@ -1,3 +1,34 @@
+#define TYPING_INDICATOR_RANGE 7
+
 /mob/proc/get_say()
+	create_typing_indicator()
 	var/msg = input(src, null, "say \"text\"") as text|null
-	say_verb(msg)
\ No newline at end of file
+	remove_typing_indicator()
+	say_verb(msg)
+
+/mob
+	var/image/typing_overlay
+	var/list/speech_bubble_recipients
+
+/mob/proc/create_typing_indicator()
+	if(typing_overlay) 
+		return
+	var/list/listening = get_hearers_in_view(TYPING_INDICATOR_RANGE, src)
+	speech_bubble_recipients = list()
+	for(var/mob/M in listening)
+		if(M.client)
+			speech_bubble_recipients.Add(M.client)
+	typing_overlay = image('icons/mob/talk.dmi', src, "default_talking", FLY_LAYER)
```

Comment:
```suggestion
	typing_overlay = image('icons/mob/talk.dmi', src, "[bubble_icon]_talking", FLY_LAYER)
```

Reply:
is bubble_icon stored at the /mob level?

Reply:
damn, its not